File: 128.longest-consecutive-sequence.cpp
// @leet start
#include <set>
#include <vector>

using namespace std;
class Solution {
public:
  int longestConsecutive(vector<int> &nums) {
    if (nums.size() == 0) {
      return 0;
    }
    set<int> s;
    for (auto n : nums) {
      s.insert(n);
    }
    bool b = false;
    int prev = 0, ml = 1, cl = 1;
    for (auto n : s) {
      if (b) {
        if (n == prev + 1) {
          cl++;
          ml = max(ml, cl);
        } else {
          cl = 1;
        }
        prev = n;
      } else {
        b = true;
        prev = n;
      }
    }
    return ml;
  }
};
// @leet end

File: 128.longest-consecutive-sequence.py
from typing import List


# @leet start
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        if not nums:
            return 0
        m = 1
        st = set(nums)
        for n in st:
            if n - 1 in st:
                continue
            cm = 0
            while n in st:
                cm += 1
                n += 1
            m = max(m, cm)
        return m


# @leet end

